### PR TITLE
TYP: misc typing cleanup in core/indexes/multi.py

### PR DESCRIPTION
**Pull request by simonjayhawkins**
pandas\core\indexes\multi.py:496: error: Need type annotation for 'arrays'  [var-annotated]
pandas\core\indexes\multi.py:722: error: Incompatible types in assignment (expression has type "List[Any]", variable has type "FrozenList")  [assignment]
pandas\core\indexes\multi.py:893: error: Incompatible types in assignment (expression has type "List[Any]", variable has type "FrozenList")  [assignment]
pandas\core\indexes\multi.py:2438: error: List item 0 has incompatible type "slice"; expected "str"  [list-item]
pandas\core\indexes\multi.py:3095: error: Unsupported left operand type for | ("None")  [operator]

### CODE REVIEW

File: pandas/core/indexes/multi.py [hunk 1]
```diff
@@ -2435,7 +2453,7 @@ def _get_partial_string_timestamp_match_key(self, key):
         if isinstance(key, str) and self.levels[0]._supports_partial_string_indexing:
             # Convert key '2016-01-01' to
             # ('2016-01-01'[, slice(None, None, None)]+)
-            key = tuple([key] + [slice(None)] * (len(self.levels) - 1))
+            key = tuple((key, *([slice(None)] * (len(self.levels) - 1))))
```
Comment (jbrockmendel) [inline pandas/core/indexes/multi.py:2456]: i find this less clear than the status quo.  what about `(key,) + (slice(None),) * num`?

Comment (WillAyd) [reply pandas/core/indexes/multi.py:2456]: Yea I agree here; if there's a way to do this without unpacking would read a lot more clearly

Alternately can be split into separate lines

Comment (simonjayhawkins) [reply pandas/core/indexes/multi.py]: updated.

**Review by WillAyd — APPROVED** [review]
lgtm

**Comment by jbrockmendel** [comment]
thanks @simonjayhawkins

**Pull request merged**